File: app/src/main/java/linusfessler/alarmtiles/TileServiceCompat.java
package linusfessler.alarmtiles;

import android.os.Build;
import android.service.quicksettings.Tile;

public class TileServiceCompat {

    private TileServiceCompat() {
    }

    public static void setSubtitle(final Tile tile, final String label, final String subtitle) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
            tile.setSubtitle(subtitle);
            return;
        }

        if (subtitle == null || subtitle.equals("")) {
            tile.setLabel(label);
        } else {
            tile.setLabel(label + "\n" + subtitle);
        }
    }
}

File: app/src/main/java/linusfessler/alarmtiles/shared/TileServiceCompat.kt
package linusfessler.alarmtiles.shared

import android.os.Build
import android.service.quicksettings.Tile

class TileServiceCompat {
    companion object {
        fun setSubtitle(tile: Tile, label: String, subtitle: String?) {
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
                tile.subtitle = subtitle
                return
            }

            if (subtitle.isNullOrBlank()) {
                tile.label = label
            } else {
                tile.label = label + "\n" + subtitle
            }
        }
    }
}